htdocs/js/apps/DragNDrop/dragndrop.js

```diff
@@ -0,0 +1,140 @@
+(function() {
+	class DragNDropBucket {
+		constructor(pgData) {
+			this.answerInputId = pgData['answerInputId'];
+			this.bucketId = pgData['bucketId'];
+			this.label = pgData['label'] || '';
+			this.removable = pgData['removable'];
+			this.bucketPool = $('.bucket_pool[data-ans="' + this.answerInputId + '"]').first()[0];
+
+			var $bucketPool = $(this.bucketPool);
+			var $newBucket = this._newBucket(
+				this.bucketId,
+				this.label,
+				this.removable,
+				$bucketPool.find('.hidden.past_answers.bucket[data-bucket-id="' + this.bucketId + '"]')
+			);
+			$bucketPool.append($newBucket);
+
+			var el = this;
+			$newBucket.find('.dd').nestable({
+				group: el.answerInputId,
+				maxDepth: 1,
+				scroll: true,
+				callback: function() {el._nestableUpdate();}
+			});
+			this._nestableUpdate();
+			this._ddUpdate();
+		}
+
+		_newBucket(bucketId, label, removable, $bucketHtmlElement) {
+			var $bucketPool = $(this.bucketPool);
+			var $newBucket = $('<div id="nestable-' + bucketId + '-container" class="dd-container"></div>');
+
+			$newBucket.attr('data-bucket-id', bucketId);
+
+			$newBucket.append($('<div class="nestable-label">' + label + '</div>'));
+			$newBucket.append($('<div class="dd" data-bucket-id="' + bucketId + '"></div>'));
+
+			if (removable != 0) {
+				$newBucket.append($('<a class="btn remove_bucket">Remove</a>'));
+			}
+
+			if ($bucketHtmlElement.find('ol.answer li').length) {
+				var $ddList = $('<ol class="dd-list"></ol>');
+				$bucketHtmlElement.find('ol.answer li').each(function(index) {
```

Comment:
The `index` parameter should be deleted as it is not used.
```suggestion
				$bucketHtmlElement.find('ol.answer li').each(function() {
```